app/uk/gov/hmrc/mobilepaye/services/MobilePayeService.scala

```diff
@@ -62,10 +62,12 @@ class MobilePayeService @Inject()(taiConnector: TaiConnector) {
         }
       }
 
-      val otherNonTaxCodeIncomes: Option[Seq[OtherIncome]] = nonTaxCodeIncomes.otherNonTaxCodeIncomes.map(income => OtherIncome.withMaybeLink(
-        name = income.getFormattedIncomeComponentType,
-        amount = income.amount.setScale(0, RoundingMode.FLOOR)
-      )) match {
+      val otherNonTaxCodeIncomes: Option[Seq[OtherIncome]] = nonTaxCodeIncomes.otherNonTaxCodeIncomes
+        .withFilter(_.incomeComponentType != BankOrBuildingSocietyInterest)
```

Comment:
Any reason for using `withFilter` instead of just `filter`?

Reply:
Good question, I did this for 2 reasons:
1. They've done exactly this in TAI
2. After looking into it, apparently: filter will take the original collection and produce a new collection, but withFilter will non-strictly (i.e. lazily) pass unfiltered values through to later map/flatMap/withFilter calls, saving a second pass through the (filtered) collection. Hence it will be more efficient when passing through to these subsequent method calls.

But maybe it doesn't matter? Happy to change it back to filter but thought why not based on what I found

Reply:
I think efficiency isn't an issue here, given the size of the collections, so really I think it comes down to familiarity and whether it will surprise people looking at the code in the future. Up to you but I'm not too worried either way.

Reply:
Good point, I prefer your view on it, so will change it back to filter, as this matches the rest of our code as well